clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -370,6 +372,155 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
         }
     }
 
+    public static PartialDefaultRecord readPartiallyFrom(DataInput input,
+                                                         ByteBuffer skipBuffer,
+                                                         long baseOffset,
+                                                         long baseTimestamp,
+                                                         int baseSequence,
+                                                         Long logAppendTime) throws IOException {
+        int sizeOfBodyInBytes = ByteUtils.readVarint(input);
+        int totalSizeInBytes = ByteUtils.sizeOfVarint(sizeOfBodyInBytes) + sizeOfBodyInBytes;
+
+        return readPartiallyFrom(input, skipBuffer, totalSizeInBytes, sizeOfBodyInBytes, baseOffset, baseTimestamp,
+            baseSequence, logAppendTime);
+    }
+
+    private static PartialDefaultRecord readPartiallyFrom(DataInput input,
+                                                          ByteBuffer skipBuffer,
```

Comment:
Would it make sense to pass this through as a byte array? That would make the call to `buffer.array()` below safer.

Reply:
Ack.